src/vs/base/browser/ui/tree/abstractTree.ts

```diff
@@ -823,6 +833,8 @@ class FindWidget<T, TFilterData> extends Disposable {
 		this.width = width;
 		this.right = clamp(this.right, 0, Math.max(0, width - 212));
 		this.elements.root.style.right = `${this.right}px`;
+		this.top = clamp(this.top, 0, this.permittedVerticalMovement);
```

Comment:
Why not just clamping this between 0 and the widget's height? That way the widget could only move vertically enough for the entire space beneath it to be revealed.